Orbitersdk/samples/ProjectApollo/src_launch/mcc.cpp

```diff
@@ -2412,11 +2412,66 @@ void MCC::drawPad(bool writetofile){
 		{
 			int hh, mm;
 			double ss;
+			char buffer2[1024];
+			std::string buffer3;
+
 			AP7MNV * form = (AP7MNV *)padForm;
-			format_time_prec(tmpbuf, form->GETI);
-			format_time_prec(tmpbuf2, form->NavChk);
+
+			if (MissionType == MTP_D)
+			{
+				buffer3 = "MANEUVER UPDATE (P30)\n";
+			}
+			else
+			{
+				buffer3 = "MANEUVER\n";
+			}
+
+			SStoHHMMSS(form->GETI, hh, mm, ss);
+			sprintf_s(buffer2, "%s PURPOSE\n%+06d HRS GETI N33\n%+06d MIN\n%+07.2f SEC\n%+07.1f DVX\n%+07.1f DVY\n%+07.1f DVZ\n", form->purpose, hh, mm, ss, form->dV.x, form->dV.y, form->dV.z);
```

Comment:
sprintf_s is a non-portable Windows only fuction.
We want NASSP to be portable so it can easily be compiled for Linux when Orbiter starts supporting it, so we want to avoid introducing new portability issues whenever we can.

You can use snprintf which does the same but in a portable way, you just need to supply a size yourself.

Reply:
Yeah I need to get in some consistency with these functions, we are using them a lot in the MCC and RTCC MFD. Originally I was only using sprintf, but sprintf_s had the advantage of causing a gauranteed CTD when there was a bug with the code (character array size not larger enough etc). I can use snprintf from now on and slowly convert all sprintf and sprintf_s if that is the right function for the job.